glmnet/tests/test_linear.py

```diff
@@ -63,6 +63,33 @@ def test_with_single_var(self):
         m = m.fit(x, y)
         self.check_r2_score(y, m.predict(x), 0.90)
 
+    def test_relative_penalties(self):
+        m1 = ElasticNet(random_state=4328)
+        m2 = ElasticNet(random_state=4328)
+        for h in self.inputs:
+            x, y = h
+            p = x.shape[1]
+
+            # m1 no relative penalties applied
+            m1.fit(x, y)
+
+            # find the nonzero indices from LASSO
+            nonzero = [ind for ind, z in enumerate(m1.coef_) if z != 0]
+
+            # get their absolute values
+            coef1abs = [np.abs(z) for z in m1.coef_]
+
+            # unpenalize those nonzero coefs
+            penalty = np.repeat(1, p)
+            penalty[nonzero] = 0
+
+            # refit the model with the unpenalized coefs
+            m2.fit(x, y, relative_penalties=penalty)
+
+            # verify that the unpenalized coef ests exceed the penalized ones
+            # in absolute value
+            coef2abs = [np.abs(z) for z in m2.coef_]
+            assert(all([z1 <= z2 for z1, z2 in zip(coef1abs, coef2abs)]))
```

Comment:
Since the estimated coef is mostly zero for both `coef1abs` and `coef2abs` and numpy.testing.assert_array_less only does strict comparison, this route probably won't work well. I don't immediately see a similar `assert_array_less_or_eq` method.